server/src/main/java/org/opensearch/action/support/replication/ReplicationModeAwareProxy.java

```diff
@@ -41,4 +91,8 @@ ReplicationMode determineReplicationMode(ShardRouting shardRouting, ShardRouting
 
         return replicationModeOverride;
     }
+
+    public ReplicationOperation.Replicas<ReplicaRequest> getPrimaryTermValidationProxy() {
+        return primaryTermValidationProxy;
+    }
```

Comment:
You can make this variable protected and avoid this

Reply:
Ack.